resources/ndstoken

```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+
+if [ -z $1 ]; then
+	echo "Usage: ndstoken <ipaddress>"
+	echo "Returns the client token"
+	exit 1
+fi
+
+ndsctl status | grep -A 6 $1' MAC' | grep Token | awk -F'Token: ' '{print $2}'
```

Comment:
The intention was to run as a command, like ndsctl, so no .sh extension.
But it is just a tool provided for convenience.
I'm sure there are 100 ways to do it :-D

==================
NoDogSplash Status
====
Version: 3.0.0
Uptime: 4h 32m 54s
Gateway Name: BlueWaveFreeWiFi
Managed interface: br-lan
Managed IP range: 0.0.0.0/0
Server listening: 10.168.1.1:2050
Client Check Interval: 200s
Preauth Idle Timeout: 120m
Auth Idle Timeout: 120m
Traffic control: no
Total download: 116461 kByte; avg: 56.90 kbit/s
Total upload: 11286 kByte; avg: 5.51 kbit/s
====
Client authentications since start: 2
Current clients: 2

Client 0
  IP: 10.168.1.68 MAC: 74:de:2b:b1:70:33
  Last Activity: Wed Aug 22 14:57:09 2018 (0s ago)
  Session Start: Wed Aug 22 12:41:39 2018 (2h 15m 30s ago)
  Session End:   -
  Token: 5906bac1
  State: Authenticated
  Download: 59238 kByte; avg: 58.29 kbit/s
  Upload:   5963 kByte; avg: 5.87 kbit/s

Client 1
  IP: 10.168.1.218 MAC: 84:38:38:8a:78:22
  Last Activity: Wed Aug 22 13:08:18 2018 (1h 48m 51s ago)
  Session Start: -
  Session End:   -
  Token: e0557d7d
  State: Preauthenticated
  Download: 0 kByte; avg: 0.00 kbit/s
  Upload:   0 kByte; avg: 0.00 kbit/s

====
Blocked MAC addresses: none
Allowed MAC addresses: N/A
Trusted MAC addresses: none
========

Reply:
Ok, your approach is fine. Mine is not that much better: `ndsctl status | grep -A 6 "IP: $1" | awk '/Token:/{print $2}'`

The .sh is a visual clue to tell people that it is a shell script. Same as .pl is for Perl scripts. What make a file a program is the executable bit.

my2cents

Reply:
Indeed. It was committed as executable so should be ok.
The script has to differentiate between eg 10.168.1.68 and 10.168.1.6
I made that mistake initially :-D
The lack of .sh and packaging so it gets stashed away in the path eg /usr/bin along with nodogsplash and ndsctl is just to make it convenient.
We could always put the functionality into ndsctl......